### PR TITLE
External Media: Improve Insert Flow

### LINKED ISSUE #15867: External Media: Improve External Media Copy/Insert Flow
At the moment we have a “Copy and Insert” button on the External Media modals. When you select some images and click that, you then see a modal showing you the images being copied.  This blocks you from moving on, while providing nothing else for you to do.

copy-insert-modal

#### Describe the solution you'd like
**Proposal** (I have no idea how feasible it is)
- Change the “Copy and Insert” button to “Select” (what the button is in the core Media Library)
- When you click Select, go immediately back to the editor, so the user isn't blocked from moving on
- Selected images are shown on the canvas in the loading state, like they would be if you drag/dropped an image on or uploaded it manually.

### PR DESCRIPTION
**Pull request by obenland**
Improves the insert flow by not taking the user out of the media modal while inserting.

#### Changes proposed in this Pull Request:

* Removes Copy/Insert Modal
* Updates Insert button to say "Select" (like in media library) and be busy while inserting.
* Updates media items to show spinner while they're being copied.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Check out this PR and run `yarn run build-extensions`
* In the Editor, insert a media block (Cover, Media & Text, Image, Gallery)
* Click "Select [Image(s)|Media]" and chose either Pexels or Google Photos.
* Based on the block, select one or more images and click "Select".
* Observe the spinner overlay over the thumbnail images and the Select button now being busy and saying "Inserting…".

**After**
image

#### Proposed changelog entry for your changes:

Not needed.

Closes #15867.
Fixes #15867.

### CODE REVIEW

**Review by WunderBart — COMMENTED** [review]
Nice UX improvement and code simplification at the same time, thanks!

I've noticed that I'm able to interact with the modal content while the copy request is pending so I decided to add a small improvement for that in https://github.com/Automattic/jetpack/pull/16081/commits/18db7348a5ed5d43d1f5410f99665af03cf44aca. I'd approve this PR but since I pushed myself it would be good for someone else to take a look.

**Comment by jeryj** [comment]
I agree with Bart - this feels a lot nicer than the Copying Modal flow 🙌

In https://github.com/Automattic/jetpack/pull/16081/commits/38fa4a9e4c7022426b43448ce563319e8db2a2f3, I added a `speak` announcement on the copying action to announce "Inserting: %s" where %s is a `join(', ')` string of all images' alt text.

File: extensions/shared/external-media/editor.scss [hunk 1]
```diff
@@ -21,39 +21,6 @@ $grid-size: 8px;
...
-
-		img {
-			width: 100%;
-			height: 100%;
-			object-fit: contain;
-		}
-	}
-}
-
 .jetpack-external-media-browser:not( .jetpack-external-media-browser__is-copying ) {
```
Comment (jeryj) [inline extensions/shared/external-media/editor.scss:24]: I wonder if we can remove this entire section, or at minimum, the `:not(...)` selector.

**Comment by jancavan** [comment]
Thanks @obenland! There's a couple more things that I think we should add to better the experience. When images are being copied:

- Thumbnails with the spinner should say "Inserting image..." to make it more obvious what's happening here in case the user misses the change in button state:

Screen Shot 2020-06-08 at 1 57 04 PM

- Search field and thumbanils should be disabled. The only thing the user should be able to do here is dismiss the modal (as an escape hatch should the copying take a long time):

Screen Shot 2020-06-08 at 1 55 45 PM

## Prototype

ezgif-6-060e0960e461

Here's a clickable prototype to get a feel for it. You can select the **2nd** and **4th** thumbnail:

https://www.figma.com/proto/ERNU68IlQZwaaDazIGfNL7/Ajax-Improve-External-Media-Flow?node-id=52%3A42&viewport=-6877%2C-4747%2C0.5&scaling=min-zoom

**Comment by jancavan** [comment]
- Modal heading also needs to change to "Insert media" once images are being copied over:

Screen Shot 2020-06-08 at 3 27 23 PM

- Should be "Select media" with lowecase `m`:

Screen Shot 2020-06-08 at 3 43 44 PM

File: extensions/shared/external-media/editor.scss [hunk 2]
```diff
@@ -71,32 +38,17 @@ $grid-size: 8px;
 	}
 }
 
+.jetpack-external-media-browser__is-copying {
+	pointer-events: none;
+}
+
```
Comment (marekhrabe) [inline extensions/shared/external-media/editor.scss:44]: While this is ok, everything will still be accessible and functional through keyboard. I think we should set our interactive elements as disabled, based on the same flag.

**Review by marekhrabe — APPROVED** [review]
Addressed:
- [x] disabled states for:  #16103
  - search input and button
  - thumbnails
  - pagination button
  - filters in google photos
- [x] append "Inserting image…" text to the loading state of thumbnail #16103

Todos from the feedback:
- [ ] add number of selected images next to the insert button (only show when multi selections are enabled)
- [ ] see if we can clean and simplify our css, as we now have much less changes for the "copying" state

I wouldn't mind if those are addressed separately, to keep the flow of PRs moving. Otherwise, this PR looks good to me and works well.

**Comment by jancavan** [comment]
@marekhrabe Would also like to add this [copy change](https://github.com/Automattic/jetpack/pull/16081#issuecomment-640925924) in, if possible. Thanks.

**Comment by obenland** [comment]
> Would also like to add this copy change in, if possible. Thanks.

Thank you! Fixed in 96ee699.

**Comment by obenland** [comment]
I noticed the modal scrolling back to top when inserting an image from below the fold. With the dedicated upload modal that was probably desired—now, not anymore. 

@marekhrabe Does that sound familiar? I've tried tracking down what might cause it but I've not come very far.

**Review by jeryj — APPROVED** [review]
:shipit:

**Pull request merged**